assume_account/ec2.tf

```diff
@@ -0,0 +1,87 @@
+resource "aws_instance" "ec2" {
+  ami           = data.aws_ami.ubuntu.id
+  instance_type = var.ec2_instance_type
+
+  subnet_id              = var.subnet_ids[0]
+  key_name               = aws_secretsmanager_secret.ec2.name
+  vpc_security_group_ids = [aws_security_group.ec2.id]
+  iam_instance_profile   = aws_iam_instance_profile.ec2.name
+
+  user_data = file("${path.module}/setup_ec2.sh")
+
+  lifecycle {
+    ignore_changes = [ami]
+  }
+
+  depends_on = [aws_security_group.ec2, aws_key_pair.ec2]
+  tags       = { Name = local.name }
+}
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name = "name"
+
+    values = [
+      "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server*",
+    ] # Needs to be as restrictive as possible
+  }
+
+  filter {
+    name = "root-device-type"
+
+    values = ["ebs"]
+  }
+  filter {
+    name = "virtualization-type"
+
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical (Ubuntu)
+}
+
+# Generate SSH key
+resource "tls_private_key" "generated_key" {
```

Comment:
Why bother creating an rsa key? SSM is much more convenient.

Reply:
So using SSM to create a key pair?